File: objetos/ControleBovinos.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package objetos;

import java.util.ArrayList;

/**
 *
 * @author anacs
 */
public class ControleBovinos {
     public static int contarVacasFerteis(ArrayList<GadoLeiteiro> listaVacas) {
        int total = 0;
        for (GadoLeiteiro vaca : listaVacas) {
            total += vaca.totalFerteis();
        }
        return total;
    }
    
    //pegar a media de litros de leite em idade fertil
    //tratar as vacas que dao menos leite
    //verificar as caracteristicas das vacas com mais leite
    
    
public static double calcularMediaLeite(ArrayList<GadoLeiteiro> vacas) {
        float totalLeite = 0;
        int totalFerteis = 0;

        for (GadoLeiteiro v : vacas) {
            if (v.totalFerteis() == 1) {
                totalLeite += v.getProducaoMensal();
                totalFerteis++;
            }
        }

        return totalFerteis == 0 ? 0 : totalLeite / totalFerteis;
    }

    public static void tratarVacasComPoucoLeite(ArrayList<GadoLeiteiro> vacas, float limiteMinimo) {
        for (GadoLeiteiro v : vacas) {
            if (v.getProducaoMensal() < limiteMinimo) {
                System.out.println("Tratando vaca " + v.identificacao + " com apenas " + v.getProducaoMensal() + " litros de leite.");
            }
        }
    }

    public static void caracteristicasDasVacasComMaisLeite(ArrayList<GadoLeiteiro> vacas) {
        float maiorProducao = 0;

        for (GadoLeiteiro v : vacas) {
            if (v.getProducaoMensal() > maiorProducao) {
                maiorProducao = v.getProducaoMensal();
            }
        }

        for (GadoLeiteiro v : vacas) {
            if (v.getProducaoMensal() == maiorProducao) {
                System.out.println("Vaca com mais leite:");
                System.out.println("- Identificação: " + v.identificacao);
                System.out.println("- Idade: " + v.idade);
                System.out.println("- Peso: " + v.getPeso());
                System.out.println("- Produção mensal: " + v.getProducaoMensal() + " litros.");
            }
        }
    }
    
}

File: python/controle_bovinos.py
class ControleBovinos:
    @staticmethod
    def contar_vacas_ferteis(lista_vacas):
        return sum(vaca.total_ferteis() for vaca in lista_vacas)

    @staticmethod
    def calcular_media_leite(vacas):
        vacas_ferteis = [v for v in vacas if v.total_ferteis() == 1]
        if not vacas_ferteis:
            return 0
        total_leite = sum(v.get_producao_mensal() for v in vacas_ferteis)
        return total_leite / len(vacas_ferteis)

    @staticmethod
    def tratar_vacas_com_pouco_leite(vacas, limite_minimo):
        for v in vacas:
            if v.get_producao_mensal() < limite_minimo:
                print(f"Tratando vaca {v.identificacao} com apenas {v.get_producao_mensal()} litros de leite.")

    @staticmethod
    def caracteristicas_das_vacas_com_mais_leite(vacas):
        if not vacas:
            return
        maior_producao = max(v.get_producao_mensal() for v in vacas)
        for v in vacas:
            if v.get_producao_mensal() == maior_producao:
                print("Vaca com mais leite:")
                print(f"- Identificação: {v.identificacao}")
                print(f"- Idade: {v.idade}")
                print(f"- Peso: {v.get_peso()}")
                print(f"- Produção mensal: {v.get_producao_mensal()} litros.")